### ISSUE TITLE
Add conditional logic to application form

### ISSUE THREAD

**Issue by avishkar58** [deep-learning-indaba/Baobab#602]
## Description
We need to add some conditional logic to the application form to show/hide questions based on the candidate's answers. This will allow us to customise the application form to the user.

## Dependencies
Depends on #601 which adds a `dependency_question_id` and `show_for_values` field to both the Section and Question models. This issue can still be implemented and tested with dummy data if #601 hasn't landed yet. 

## Tasks
Update the [ApplicationForm](https://github.com/deep-learning-indaba/Baobab/blob/develop/webapp/src/pages/applicationForm/components/ApplicationForm.js) component to filter Sections and Questions according to the following criteria:
- If dependency_question_id is null, keep the section/question
- If dependency_question_id is not null, retrieve the answer to the dependency question. 
  - If the answer is null, don't show the section/question. 
  - If the answer is not null, show the section/question if the answer's value is contained in the `show_for_values` JSON list. 

The above logic is very similar to the [Registration component](https://github.com/deep-learning-indaba/Baobab/blob/ed9ef8986eeb5d725ce24cffdbbec15b7224c83a/webapp/src/pages/registration/components/RegistrationComponent.js#L495), except here we use a "show list" rather than a single "hide value".

**Comment by avishkar58** [comment]
Hey @TmanTman - just letting you know that issue #601 (PR #622) has been merged into develop now incase you were waiting for that to implement this issue.

**Comment by TmanTman** [comment]
Thanks, will test and submit PR

**Comment by TmanTman** [comment]
After pulling the latest version of the develop branch and running

`docker-compose run web python ./api/run.py db upgrade --directory api/migrations`

I got the error

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column question.depends_on_question_id does not exist
LINE 1: ...ns, question.is_required AS question_is_required, question.d...
```

Any ideas what could cause this? I'll also try and follow up with @MrJordanWilliam

**Comment by SMHall94** [comment]
@avishkar58, @amritpurshotam Do you perhaps have any ideas? :)

**Comment by avishkar58** [comment]
Hey, which migration is this occuring on?

**Comment by amritpurshotam** [comment]
@TmanTman I've checked out the latest develop branch and re-ran all the migrations (from a clean db) and it completed successfully. Can you make sure you've got the latest version of the develop branch?

cc @SMHall94 @avishkar58

**Comment by TmanTman** [comment]
@amritpurshotam Thanks not sure what state my repo was in before last evening. Working now. Made some adjustments and still working on the feature.

**Comment by avishkar58** [comment]
Hi @TmanTman - are you still working on this issue?

**Comment by TmanTman** [comment]
I am.

My initial work failed to consider both the saved answer on the question,
and the current answers (kept in a separate variable). Just accounting for
that now.

**Comment by TmanTman** [comment]
I've updated the PR. I'm seeing weird behaviour on StepZilla. As soon as the answers load and section is dynamically inserted (because if it's dependency on another question's answer) the progress bar is extended.  The class that is applied to the last step is wrong, breaking the progress bar:
<img width="1408" alt="Screenshot 2020-03-17 at 23 15 37" src="https://user-images.githubusercontent.com/3293478/76902813-729ce980-68a5-11ea-8fb3-70c4639f5114.png">

I'll have to look into why that's happening

**Comment by SMHall94** [comment]
@TmanTman Hi there :) I hope the isolation isn't getting you down :) Have you made any progress with regards to the progress bar issue? :)

**Comment by avishkar58** [comment]
Progress bar seemed to work fine in my testing - closing
